nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/ConsumeAzureEventHub.java

```diff
@@ -91,7 +91,8 @@
 })
 public class ConsumeAzureEventHub extends AbstractSessionFactoryProcessor {
 
-    private static final String FORMAT_STORAGE_CONNECTION_STRING = "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s";
+    private static final String FORMAT_STORAGE_CONNECTION_STRING_FOR_ACCOUNT_KEY = "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s";
+    private static final String FORMAT_STORAGE_CONNECTION_STRING_FOR_SAS_TOKEN = "BlobEndpoint=https://%s.blob.core.windows.net/;SharedAccessSignature=%s";
```

Comment:
Putting `blob.core.windows.net` explicitly in this means that this isn't going to work for any special regions including US Government, China, and Germany. Probably need to find a way to add endpoint suffix support to this.

Reply:
@jfrazee At the moment ConsumeAzureEventHub processor does not support special regions, so this is why it is not included in this PR. During AzureCheckpointLeaseManager initialisation CloudStorageAccount is created by the parse(storageConnectionString). Parse is using tryConfigureServiceAccount() method to instantiate CloudStrageAccount using public CloudStorageAccount(StorageCredentials storageCredentials, StorageUri blobStorageUri, StorageUri queueStorageUri, StorageUri tableStorageUri, StorageUri fileStorageUri) constructor. This constructor sets endPointSuffix to null so the url always be "core windows.net"
It is a good idea though and can be added by a separate request.

Reply:
https://issues.apache.org/jira/browse/NIFI-8677 has been opened for endpoint suffix support.